wallet-new/src/Cardano/Wallet/Launcher.hs

```diff
@@ -0,0 +1,290 @@
+import           Pos.Wallet.Web.State (askWalletDB, askWalletSnapshot,
+                     flushWalletStorage, getWalletAddresses)
+import           Pos.Wallet.Web.State.Acidic (closeState, openState)
+import           Pos.Wallet.Web.Tracking (eskToWalletDecrCredentials,
+                     syncWallet)
+import           Pos.Web.Types (TlsParams (..))
+import           Pos.Worker.Update (updateTriggerWorker)
+import           Pos.WorkMode (RealModeContext (..))
+
+import qualified Cardano.Wallet.Kernel.Keystore as Keystore
+import qualified Cardano.Wallet.Kernel.Mode as Kernel.Mode
+import qualified Cardano.Wallet.Server.Plugins as Plugins
+
+
+-- | start a node from node arguments
+startCoreNode
+    :: CommonNodeArgs
+    -> NodeArgs
+    -> LoggingParams
+    -> IO ()
+startCoreNode cArgs nArgs lArgs = do
+    let blPath = AssetLockPath <$> cnaAssetLockPath cArgs
+
+    loggerBracket lArgs . logException (lpDefaultName lArgs) $
+        withCompileInfo $ withConfigurations blPath (getConfigOpts cArgs) $ \_ pm -> do
+            (params, Just ssc) <- getNodeParams (lpDefaultName lArgs) cArgs nArgs
+            runNodeReal pm params ssc [updateTriggerWorker]
+
+
+-- | Runs an wallet node plus its wallet backend API.
+startWalletNode
+    :: NodeArgs
+    -> WalletStartupOptions
+    -> LoggingParams
+    -> IO ()
+startWalletNode nArgs wOpts lArgs = do
+    let cArgs  = wsoNodeArgs wOpts
+    let blPath = AssetLockPath <$> cnaAssetLockPath cArgs
+
+    loggerBracket lArgs . logException (lpDefaultName lArgs) $ do
+        logInfo "[Attention] Software is built with the wallet backend"
+        withCompileInfo $ withConfigurations blPath (getConfigOpts cArgs) $ \ntpConfig pm -> do
+            (nodeParams, Just sscParams) <- getNodeParams (lpDefaultName lArgs) cArgs nArgs
+            printInfoOnStart cArgs ntpConfig
+            logInfo "Wallet is enabled!"
+            case wsoWalletBackendParams wOpts of
+              WalletLegacy legacyParams ->
+                actionWithWallet pm sscParams nodeParams ntpConfig legacyParams
+              WalletNew newParams ->
+                actionWithNewWallet pm sscParams nodeParams newParams
+
+
+-- | 'WalletWebMode' headless one-shot runner. If differs from `runWRealMode`
+-- by several aspects:
+--
+--     - It runs the given action and returns, whereas `runWRealMode`
+--       actually starts a server waiting for an explicit shutdown.
+--
+--     - It doesn't run the full diffusion layer, so it's made only for
+--       isolate actions on the wallet.
+--
+--     - It only requires 'external' configurations parameters, as they'd be
+--       passed to a node and it handles the internal logic. This makes it much
+--       easier to use as one can merely specify a configuration file, a few
+--       paths and magic happens.
+runWWebMode
+    :: forall a
+    .  CommonNodeArgs
+    -> NodeArgs
+    -> WalletBackendParams
+    -> ((HasConfigurations, HasCompileInfo) => WalletWebMode a)
```

Comment:
Possibly, we could make this:

```hs
     -> ((HasConfigurations, HasCompileInfo) => Diffusion WalletWebMode -> WalletWebMode a)
```

Though I don't think we want to spend any more time on this knowing that the new data-layer will pretty much makes this obsolete.